shell/common/platform_util_win.cc

```diff
@@ -244,10 +244,17 @@ std::string OpenExternalOnWorkerThread(
   base::string16 escaped_url = L"\"" + base::UTF8ToUTF16(url.spec()) + L"\"";
   base::string16 working_dir = options.working_dir.value();
 
-  if (reinterpret_cast<ULONG_PTR>(
-          ShellExecuteW(nullptr, L"open", escaped_url.c_str(), nullptr,
-                        working_dir.empty() ? nullptr : working_dir.c_str(),
-                        SW_SHOWNORMAL)) <= 32) {
+  LONGLONG open_call_res = reinterpret_cast<LONGLONG>(ShellExecuteW(
+      nullptr, L"open", escaped_url.c_str(), nullptr,
+      working_dir.empty() ? nullptr : working_dir.c_str(), SW_SHOWNORMAL));
+  if (options.fallback_to_user && open_call_res == SE_ERR_NOASSOC) {
+    const OPENASINFO info = {escaped_url.c_str(), nullptr,
+                             OAIF_ALLOW_REGISTRATION | OAIF_REGISTER_EXT |
+                                 OAIF_EXEC | OAIF_FILE_IS_URI};
```

Comment:
why `OAIF_EXEC` ? we are only delegating to find applications that can open the url, there is no file to execute.

Can this just be `OAIF_ALLOW_REGISTRATION | OAIF_REGISTER_EXT | OAIF_FILE_IS_URI` ?

Reply:
No. `OAIF_EXEC` is required for Windows to display the proper program select dialog, and then open the link with that selected program. Without this flag, the user only sees a message from Windows telling them they can manually change the file association through Settings > Apps > Default Apps.